mona_sdk/authentication.py

```diff
@@ -313,19 +311,32 @@ def inner(*args, **kwargs):
             # args[0] is the current mona_client instance.
             mona_client = args[0]
 
+            # If len(args) < 1, the wrapped function does not have args to log (neither
+            # messages nor config)
+            should_log_args = len(args) > 1 and mona_client.save_failed_messages_logs
+
             if not is_authenticated(mona_client.api_key):
                 get_logger().warn("Mona's client is not authenticated")
+                if should_log_args:
+                    get_logger().error(
```

Comment:
this is a code repetition, can you use _handle_authentication_error?

Reply:
oops